File: app/src/main/java/com/kazucocoa/droidtesthelper/HandleBroadcastReceiver.java
package com.kazucocoa.droidtesthelper;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.support.annotation.NonNull;

public class HandleBroadcastReceiver extends BroadcastReceiver {

    private static final String TAG = HandleBroadcastReceiver.class.getSimpleName();

    @Override
    public void onReceive(Context context, Intent intent) {
        if (MainActivity.hasExtraRegardingAccountType(intent)) {
            Intent buildLaunchingMainActivityIntent =
                    MainActivity.buildLaunchingMainActivityIntent(context, intent);
            launchMainActivity(context, buildLaunchingMainActivityIntent);
        }

        if (HandleLocaleActivity.hasExtraRegardingLocal(intent)) {
            Intent handleLocalActivityIntent =
                    HandleLocaleActivity.buildLaunchHandleLocalActivityIntent(context, intent);
            launchHandleLocalActivity(context, handleLocalActivityIntent);
        }

        if (HandleAnimations.hasExtraRegardingAnimation(intent)) {
            HandleAnimations.enableAnimationsWithIntent(intent);
        }
    }

    public void launchMainActivity(@NonNull Context context, @NonNull Intent intent) {
        context.startActivity(intent);
    }

    public void launchHandleLocalActivity(@NonNull Context context, @NonNull Intent intent) {
        context.startActivity(intent);
    }
}

File: app/src/main/java/com/kazucocoa/droidtesthelper/HandleBroadcastReceiver.kt
package com.kazucocoa.droidtesthelper

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

import com.kazucocoa.droidtesthelperlib.HandleAnimations

class HandleBroadcastReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        if (MainActivity.hasExtraRegardingAccountType(intent)) {
            MainActivity.buildLaunchingMainActivityIntent(context, intent).let { launchMainActivity(context, it) }
        }

        if (HandleLocaleActivity.hasExtraRegardingLocal(intent)) {
            HandleLocaleActivity.buildLaunchHandleLocalActivityIntent(context, intent).let { launchHandleLocalActivity(context, it) }
        }

        if (HandleAnimations.hasExtraRegardingAnimation(intent)) {
            HandleAnimations.enableAnimationsWithIntent(intent)
        }
    }

    private fun launchMainActivity(context: Context, intent: Intent) {
        context.startActivity(intent)
    }

    private fun launchHandleLocalActivity(context: Context, intent: Intent) {
        context.startActivity(intent)
    }

    companion object {

        private val TAG = HandleBroadcastReceiver::class.java.simpleName

        const val RECEIVER_NAME = "DroidTestHelper"
    }
}
